src/Microsoft.AspNet.Razor/Parser/TokenizerBackedParser.cs

```diff
@@ -75,6 +76,39 @@ protected void Initialize(SpanBuilder span)
             }
         }
 
+        protected TSymbol Lookahead(int count = 1)
+        {
+            if (count == 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
+
+            // We add 1 in order to store the current symbol.
+            var symbols = new TSymbol[count + 1];
+            var currentSymbol = CurrentSymbol;
+
+            symbols[0] = currentSymbol;
+
+            // We need to lookforward "count" many times.
+            for(var i = 1; i <= count; i++)
+            {
+                NextToken();
+                symbols[i] = CurrentSymbol;
+            }
+
+            // We don't want to modify where the Tokenizer is pointing. Restore it to what it was before the lookahead.
```

Comment:
- remove first sentence -- desires are meaningless to a computer and to most readers.
- clarify remaining sentence: `Restore Tokenizer's location to where it was pointing before the look-ahead."